pkg/v1/tarball/write.go

```diff
@@ -161,6 +200,10 @@ func MultiRefWrite(refToImage map[name.Reference]v1.Image, w io.Writer) error {
 			if err := writeTarEntry(tf, layerFiles[i], r, blobSize); err != nil {
 				return err
 			}
+
+			// at this point, we have the config file, update the info
+			complete += blobSize
```

Comment:
Rather than just sending updates at the end of each blob write, we could add a wrapper to the `tar.Writer` that counts bytes as we write them? Then the updates would have much higher resolution.

Reply:
For sure. In the project I am working on to consume this, someone did precisely that. I will update the PR to provide that.

Reply:
Done. Take a look and let me know what you think.